modules/apps/callbacks/types/callbacks.go

```diff
@@ -0,0 +1,81 @@
+package types
+
+import (
+	porttypes "github.com/cosmos/ibc-go/v7/modules/core/05-port/types"
+	ibcexported "github.com/cosmos/ibc-go/v7/modules/core/exported"
+)
+
+// PacketUnmarshalerIBCModule is an interface that combines the IBCModule and PacketDataUnmarshaler
+// interfaces to assert that the underlying application supports both.
+type PacketUnmarshalerIBCModule interface {
```

Comment:
I think we can maybe rename. I would want a name that indicates that the module which implements this interface fully supports the callback middleware. So something like `CallbackCompatibleIBCModule` but I think there's probably some better name I am not thinking of